compysition/queue.py

```diff
@@ -73,14 +73,19 @@ def __init__(self, size=0):
         self.error = _InternalQueuePool(size=size)
         self.logs = _InternalQueuePool(size=size, placeholder=uuid().get_hex())
 
-    def list_all_queues(self):
-        queue_list = self.inbound.values() + self.outbound.values() + self.error.values() + self.logs.values()
-        return queue_list
+    def _list_of_queue_lists(self):
+        return [
+            self.inbound,
+            self.outbound,
+            self.error,
+            self.logs
+        ]
 
     def join(self):
         """**Blocks until all queues in the pool are empty.**"""
-        for queue in self.list_all_queues():
-            queue.wait_until_empty()
+        for lst in self._list_of_queue_lists():
```

Comment:
I don't like this change as I am removing a function all together and it feels "hacky" (and haven't fully tested).  Though the theory was to iterate each list rather than iterating a **new** merged list.